### PR TITLE
fix container build

### PR DESCRIPTION
**Pull request by BenTheElder**
- install npm & postcss-cli in docker image (required by the new docsy theme)
- match container build and non-container build hugo command (add --minify)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -64,7 +64,7 @@ container-image:
 		--build-arg HUGO_VERSION=$(HUGO_VERSION)
 
 container-build: module-check
-	$(CONTAINER_RUN) $(CONTAINER_IMAGE) hugo
+	$(CONTAINER_RUN) $(CONTAINER_IMAGE) hugo --minify
```
Comment (mrbobbytables) [inline Makefile:67]: This should be double checked, minify was causing problems for svgs :(

It was fixed in the upstream [minify library](https://github.com/tdewolff/minify) as of 2.7.4, but to my knowledge has not been included in Hugo yet. I had to pull it out of the contrib site config  for breaking the kubernetes logo.
ref: https://github.com/kubernetes-sigs/contributor-site/pull/131
ref: https://github.com/gohugoio/hugo/issues/6892

**Comment by kbhawkey** [comment]
Hi @BenTheElder .
Would you open an issue for these changes? Thanks!

**Comment by rlenferink** [comment]
<details>
  <summary>make container-image</summary>

```bash
[root@centos8 website]# make container-image
docker build . \
        --network=host \
        --tag kubernetes-hugo \
        --build-arg HUGO_VERSION=0.70.0
Sending build context to Docker daemon  3.584kB
Step 1/7 : FROM alpine:latest
 ---> a24bb4013296
Step 2/7 : LABEL maintainer="Luc Perkins <lperkins@linuxfoundation.org>"
 ---> Running in 69ba176bde5c
Removing intermediate container 69ba176bde5c
 ---> 34815107636a
Step 3/7 : RUN apk add --no-cache     curl     git     openssh-client     rsync     build-base     libc6-compat
 ---> Running in f074d4cc7dec
fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/community/x86_64/APKINDEX.tar.gz
(1/37) Upgrading musl (1.1.24-r8 -> 1.1.24-r9)
(2/37) Installing libgcc (9.3.0-r2)
(3/37) Installing libstdc++ (9.3.0-r2)
(4/37) Installing binutils (2.34-r1)
(5/37) Installing libmagic (5.38-r0)
(6/37) Installing file (5.38-r0)
(7/37) Installing gmp (6.2.0-r0)
(8/37) Installing isl (0.18-r0)
(9/37) Installing libgomp (9.3.0-r2)
(10/37) Installing libatomic (9.3.0-r2)
(11/37) Installing libgphobos (9.3.0-r2)
(12/37) Installing mpfr4 (4.0.2-r4)
(13/37) Installing mpc1 (1.1.0-r1)
(14/37) Installing gcc (9.3.0-r2)
(15/37) Installing musl-dev (1.1.24-r9)
(16/37) Installing libc-dev (0.7.2-r3)
(17/37) Installing g++ (9.3.0-r2)
(18/37) Installing make (4.3-r0)
(19/37) Installing fortify-headers (1.1-r0)
(20/37) Installing patch (2.7.6-r6)
(21/37) Installing build-base (0.5-r2)
(22/37) Installing ca-certificates (20191127-r4)
(23/37) Installing nghttp2-libs (1.41.0-r0)
(24/37) Installing libcurl (7.69.1-r0)
(25/37) Installing curl (7.69.1-r0)
(26/37) Installing expat (2.2.9-r1)
(27/37) Installing pcre2 (10.35-r0)
(28/37) Installing git (2.26.2-r0)
(29/37) Installing libc6-compat (1.1.24-r9)
(30/37) Installing openssh-keygen (8.3_p1-r0)
(31/37) Installing ncurses-terminfo-base (6.2_p20200523-r0)
(32/37) Installing ncurses-libs (6.2_p20200523-r0)
(33/37) Installing libedit (20191231.3.1-r0)
(34/37) Installing openssh-client (8.3_p1-r0)
(35/37) Installing libacl (2.2.53-r0)
(36/37) Installing popt (1.16-r7)
(37/37) Installing rsync (3.1.3-r3)
Executing busybox-1.31.1-r16.trigger
Executing ca-certificates-20191127-r4.trigger
OK: 228 MiB in 50 packages
Removing intermediate container f074d4cc7dec
 ---> c4ee6d31d490
Step 4/7 : ARG HUGO_VERSION
 ---> Running in cd72b0e2b39d
Removing intermediate container cd72b0e2b39d
 ---> 611b1b423989
Step 5/7 : RUN mkdir -p /usr/local/src &&     cd /usr/local/src &&     curl -L https://github.com/gohugoio/hugo/releases/download/v${HUGO_VERSION}/hugo_extended_${HUGO_VERSION}_Linux-64bit.tar.gz | tar -xz &&     mv hugo /usr/local/bin/hugo &&     addgroup -Sg 1000 hugo &&     adduser -Sg hugo -u 1000 -h /src hugo
 ---> Running in 05b65fe7ad74
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   650  100   650    0     0   1981      0 --:--:-- --:--:-- --:--:--  1981
100 13.8M  100 13.8M    0     0  6666k      0  0:00:02  0:00:02 --:--:-- 9776k
Removing intermediate container 05b65fe7ad74
 ---> 2707a22e25ea
Step 6/7 : WORKDIR /src
 ---> Running in 6fa6a650cfd3
Removing intermediate container 6fa6a650cfd3
 ---> c2e82c84da78
Step 7/7 : EXPOSE 1313
 ---> Running in 15c22258d42c
Removing intermediate container 15c22258d42c
 ---> 89edb4a047c4
Successfully built 89edb4a047c4
Successfully tagged kubernetes-hugo:latest
```
</details>

<details>
  <summary>make container-build</summary>

```bash
[root@centos8 website]# make container-build
docker run --rm --interactive --tty --volume /mnt/projects/CNCF/k8s/website:/src kubernetes-hugo hugo
Total in 121509 ms
Error: Error building site: POSTCSS: failed to transform "scss/main.css" (text/css): resource "scss/scss/main.scss_4853eb546e7a6c0898ed71feae7357c0" not found in file cache
make: *** [Makefile:67: container-build] Error 255
```
</details>

<details>
  <summary>make container-serve</summary>

```bash
[root@centos8 website]# make container-serve
docker run --rm --interactive --tty --volume /mnt/projects/CNCF/k8s/website:/src --mount type=tmpfs,destination=/src/resources,tmpfs-mode=0755 -p 1313:1313 kubernetes-hugo hugo server --buildFuture --bind 0.0.0.0

                   |  EN  |  ZH  |  KO  |  JA  |  FR  |  IT  |  DE  |  ES  |  PT  |  ID  |  RU  |  VI  |  PL  |  UK
-------------------+------+------+------+------+------+------+------+------+------+------+------+------+------+-------
  Pages            | 1106 |  766 |  375 |  242 |  243 |   56 |  104 |  164 |   51 |  190 |  115 |   63 |   54 |   64
  Paginator pages  |   33 |    4 |    0 |    0 |    0 |    0 |    0 |    0 |    0 |    0 |    0 |    0 |    0 |    0
  Non-page files   |  435 |  356 |  218 |  238 |   60 |   25 |   21 |   33 |   25 |   80 |   29 |   13 |    9 |   31
  Static files     | 1092 | 1092 | 1092 | 1092 | 1092 | 1092 | 1092 | 1092 | 1092 | 1092 | 1092 | 1092 | 1092 | 1092
  Processed images |    0 |    0 |    0 |    0 |    0 |    0 |    0 |    0 |    0 |    0 |    0 |    0 |    0 |    0
  Aliases          |    8 |    4 |    1 |    0 |    0 |    0 |    0 |    0 |    0 |    2 |    0 |    0 |    0 |    0
  Sitemaps         |    2 |    1 |    1 |    1 |    1 |    1 |    1 |    1 |    1 |    1 |    1 |    1 |    1 |    1
  Cleaned          |    0 |    0 |    0 |    0 |    0 |    0 |    0 |    0 |    0 |    0 |    0 |    0 |    0 |    0

Built in 102329 ms
Watching for changes in /src/{archetypes,assets,content,data,i18n,layouts,static,themes}
Watching for config changes in /src/config.toml, /src/themes/docsy/config.toml
Environment: "development"
Serving pages from memory
Running in Fast Render Mode. For full rebuilds on change: hugo server --disableFastRender
Web Server is available at http://localhost:1313/ (bind address 0.0.0.0)
Press Ctrl+C to stop
```
</details>

The `make container-build` fails for me as well. 

With your addition the build continues to fail because autoprefixer is missing.

<details>
  <summary>make container-build</summary>

Building sites … Error: Loading PostCSS Plugin failed: Cannot find module 'autoprefixer'
Require stack:
- /src/noop.js

(@/src/themes/docsy/postcss.config.js)
    at load (/src/node_modules/postcss-load-config/src/plugins.js:28:11)
    at /src/node_modules/postcss-load-config/src/plugins.js:53:16
    at Array.map (<anonymous>)
    at plugins (/src/node_modules/postcss-load-config/src/plugins.js:52:8)
    at processResult (/src/node_modules/postcss-load-config/src/index.js:33:14)
    at /src/node_modules/postcss-load-config/src/index.js:94:14
    at async Promise.all (index 0)
Total in 106879 ms
</details>

The suggestion of @mrbobbytables is working: https://github.com/kubernetes-sigs/contributor-site/blob/master/Dockerfile#L14

File: Dockerfile [hunk 2]
```diff
@@ -14,7 +14,9 @@ RUN apk add --no-cache \
     openssh-client \
     rsync \
     build-base \
-    libc6-compat
+    libc6-compat \
+    npm && \
+    npm install postcss-cli
```
Comment (rlenferink) [inline Dockerfile:19]: Autoprefixer is missing as well here. See my suggestion ([copy from contributor-site](https://github.com/kubernetes-sigs/contributor-site/blob/028e025e3bcd3af87004bf6e7390c69d1f4a76b1/Dockerfile#L15-L18) which also includes a comment on why the components are installed).

```suggestion
    npm
    
# Required for PostCSS
RUN npm install -G \
    autoprefixer \
    postcss-cli
```

**Comment by kbhawkey** [comment]
https://github.com/google/docsy#prerequisites

**Comment by BenTheElder** [comment]
Sorry I was out (work holiday / USA) and then this got lost in the noise. Coming back to it now.

**Comment by BenTheElder** [comment]
Filed https://github.com/kubernetes/website/issues/22537 to track and added autoprefixer to the install

**Comment by rlenferink** [comment]
NP @bentheelder

/lgtm

**Comment by celestehorgan** [comment]
@BenTheElder I've pulled down your branch and am testing it locally – one tangential note:

We need to change [L9](https://github.com/kubernetes/website/pull/22258/files#diff-3254677a7917c6c01f55212f86c57fbfR9), as @lucperkins  is leaving the LF tomorrow 😢.

I'm not sure who the best replacement is however.

**Comment by BenTheElder** [comment]
Oh :(

Maybe we should do that in a follow up?
IIRC this line could also just be removed without really affecting anything.

**Comment by celestehorgan** [comment]
@BenTheElder yeah we can do this via follow-up :( :( 

She builds! Thanks for doing this btw!

/approve

**Pull request merged**